### PR TITLE
Bug fix for #761 and color enhance when temperature in flux

### PR DESCRIPTION
**Pull request by ciotlosm**
- In Flux temperature status with color
- Enforce min value when no target_temp set in configuration
- Correct triggering temperature when min is reached and user clicks extra #761

### CODE REVIEW

File: src/components/ha-climate-control.html [hunk 1]
```diff
@@ -14,6 +14,9 @@
         @apply(--layout-horizontal);
         @apply(--layout-justified);
       }
+      .in-flux.target-temperature {
+        color: red;
```
Comment (c727) [inline src/components/ha-climate-control.html]: color: var(--google-red-500);

Comment (ciotlosm) [reply src/components/ha-climate-control.html]: Fixed

File: src/components/ha-climate-control.html [hunk 2]
```diff
@@ -67,14 +70,30 @@
       },
     };
   }
+  temperatureStateInFlux(inFlux) {
+    var tempElement = this.root.querySelector('.target-temperature');
```
Comment (andrey-git) [inline src/components/ha-climate-control.html]: Change `target-temperature` to an ID - then you can access it via this.$

File: src/components/ha-climate-control.html [hunk 3]
```diff
@@ -67,14 +70,30 @@
...
+    if (inFlux) {
```
Comment (andrey-git) [inline src/components/ha-climate-control.html]: Use
`tempElement.classList.toggle('in-flux', inFlux);` instead

File: src/components/ha-climate-control.html [hunk 4]
```diff
@@ -48,7 +51,7 @@
 
   static get properties() {
     return {
-      value: {
+      last_changed: {
```
Comment (andrey-git) [inline src/components/ha-climate-control.html]: last_changed is changed internally - no need to add observer to it - you can just call the needed function.

`value` if passed to this component, so it needs to be declared and observed

**Review by andrey-git — CHANGES_REQUESTED** [review]
General comment: Use `let` and `const` instead of `var` in new code.

File: src/components/ha-climate-control.html (hunk 4 above)
Comment (ciotlosm) [reply src/components/ha-climate-control.html]: The observer doesn't function as it should. It generates a bug when reaching end of interval. 
Should I will just add method call at end of each increment method?

**Comment by ciotlosm** [comment]
Updated all of the above

File: src/components/ha-climate-control.html [hunk 5]
```diff
@@ -84,21 +98,20 @@
     } else {
       this.value = this.min;
     }
+    this.temperatureStateInFlux(true);
+    this.valueChanged();
   }
   valueChanged() {
```
Comment (andrey-git) [inline src/components/ha-climate-control.html]: The bug was due to the following flow:
* The value is changed to max. last_updated is changed and observer is fired and sets timeout to 2010ms in the future.
* incrementValue is called. last_updated is changed but observer is *not* called.
* 2010ms after the first click callback is called but 2s didn't pass from last click.

To fix this - last_changed should only be changed when value changes.

Comment (ciotlosm) [reply src/components/ha-climate-control.html]: I will try the suggestion

Comment (ciotlosm) [reply src/components/ha-climate-control.html]: You were right, the problem was last_updated. Fixed that now based on the above suggestion.

**Comment by ciotlosm** [comment]
Also modified rest of file to use `const` instead of `var` where it was possible.

File: src/components/ha-climate-control.html [hunk 6]
```diff
@@ -67,38 +70,47 @@
...
-    var newval = this.value + this.step;
-    this.last_changed = Date.now();
+    const newval = this.value + this.step;
     if (newval <= this.max) {
-      this.value = newval;
+      // If no initial target_temp
+      // this forces control to start
+      // from the min configured instead of 0
+      if (newval <= this.min) {
+        this.value = this.min;
```
Comment (andrey-git) [inline src/components/ha-climate-control.html]: I think
```
this.last_changed = Date.now();
this.temperatureStateInFlux(true);
```
should apply to this case too.

File: src/components/ha-climate-control.html [hunk 7]
```diff
@@ -67,38 +70,47 @@
...
+      }
     } else {
       this.value = this.max;
     }
   }
   decrementValue() {
-    var newval = this.value - this.step;
-    this.last_changed = Date.now();
+    const newval = this.value - this.step;
     if (newval >= this.min) {
```
Comment (andrey-git) [inline src/components/ha-climate-control.html]: I think the following case is unhandled:
min=10
value=10.5
step=1

the result is less than min, but setting value=min still changes value.
Everything would still work, but the 2s time won't get restarted.

Comment (ciotlosm) [reply src/components/ha-climate-control.html]: I improved the detection of value changed to put temperature color correctly and trigger last_changed trigger

File: src/components/ha-climate-control.html (hunk 6 above)
Comment (ciotlosm) [reply src/components/ha-climate-control.html]: I improved the detection of value changed to put temperature color correctly and trigger last_changed trigger

**Comment by ciotlosm** [comment]
@andrey-git Are there any other changes I missed in the PR?

**Review by andrey-git — APPROVED** [review]

**Comment by ciotlosm** [comment]
Will this make it in 0.61?

**Comment by ciotlosm** [comment]
@andrey-git anything else required to get this merged?

**Comment by andrey-git** [comment]
Sorry for the delay.

**Comment by ciotlosm** [comment]
@andrey-git thank you

**Pull request merged**